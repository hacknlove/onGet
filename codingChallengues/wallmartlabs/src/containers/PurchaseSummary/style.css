.purchase-summary {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  float: right;
  min-width: 235px;
  width: 40%;
  max-width: 300px;
  border: solid 1px #ccc;
  padding: 16px;
  border-radius: 0 0 5px 5px;
  box-shadow: 0 2px 3px 0 rgba(0,0,0,.16);
  line-height: 24px;
}
