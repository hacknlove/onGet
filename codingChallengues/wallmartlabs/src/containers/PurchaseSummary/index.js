import React from 'react';
import { useOnGet } from 'onget'
import Row from '../../components/Row';
import ExpandableItem from '../../components/ExpandableItem';
import DiscountForm from '../../components/DiscountForm';
import ItemDetails from '../../components/ItemDetails';
import ToolTip from '../../components/ToolTip';
import './style.css';


export default function PurchaseSummary ({ url }) {
  const purchaseData = useOnGet(url)
  if (!purchaseData) {
    return null
  }

  return (
    <div className="purchase-summary">
      <Row title="Subtotal" figure={purchaseData.pricing.subtotal} />
      <Row
        title="Pickup savings"
        figure={purchaseData.pricing.savings}
        less={true}
        toolTip={
          <ToolTip
            title="Pickup savings"
            desc="Picking up your order in the store helps cut costs, and we pass the savings on to you."
          />
        }
      />
      {purchaseData.pricing.discount > 0 ? (
        <Row
          title="Discount"
          figure={purchaseData.pricing.discount}
          less={true}
        />
      ) : null}
      <Row title="Est. taxes and fees" figure={purchaseData.pricing.tax}>
        <div>(Based on {purchaseData.pricing.zip})</div>
      </Row>
      <Row
        total={true}
        title="Est. total"
        figure={purchaseData.pricing.total}
      />
      <ExpandableItem
        openPrefix={'See'}
        closePrefix={'Hide'}
        title={'item details'}
      >
        <ItemDetails {...purchaseData.itemDetails} />
      </ExpandableItem>
      <ExpandableItem
        openPrefix={'Apply'}
        closePrefix={'Hide'}
        title={'promo code'}
      >
        <DiscountForm url={url} />
      </ExpandableItem>
    </div>
  );
}
