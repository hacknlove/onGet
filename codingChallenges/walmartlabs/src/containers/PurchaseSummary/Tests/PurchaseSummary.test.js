import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PurchaseSummary from '../index';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { set } from 'onget'

set('dotted://purchaseData', {
  pricing: {
    subtotal: 102.96,
    previousPrice: 120.99,
    savings: 3.85,
    discount: 0.0,
    tax: 8.92,
    total: 108.95,
    zip: 94702
  },
  itemDetails: {
    itemName: 'OFM Essentials Racecar-Style Leather Gaming Chair, Green',
    pictureURL:
    "https://i5.walmartimages.com/asr/79916c35-3293-48ff-acff-be325b2e4c84_2.ef5b1bd1451de6b63a7c20d2485d56c6.jpeg?odnHeight=100&odnWidth=100&odnBg=FFFFFF",
    quantity: 1,
    price: 99.11,
    prevPrice: 102.96,
  }
})

configure({ adapter: new Adapter() });
describe('purchase summary container', () => {
  it('matches snapshot', function() {
    let wrapper = shallow(<PurchaseSummary url="dotted://purchaseData"/>);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('renders without crashing', function() {
    let wrapper = shallow(<PurchaseSummary url="dotted://purchaseData"/>);
    console.log(wrapper.debug());
  });
});
