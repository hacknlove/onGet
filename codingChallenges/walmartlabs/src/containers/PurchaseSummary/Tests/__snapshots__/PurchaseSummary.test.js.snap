// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`purchase summary container matches snapshot 1`] = `
<div
  className="purchase-summary"
>
  <Row
    figure={102.96}
    title="Subtotal"
  />
  <Row
    figure={3.85}
    less={true}
    title="Pickup savings"
    toolTip={
      <ToolTip
        desc="Picking up your order in the store helps cut costs, and we pass the savings on to you."
        title="Pickup savings"
      />
    }
  />
  <Row
    figure={8.92}
    title="Est. taxes and fees"
  >
    <div>
      (Based on
      94702
      )
    </div>
  </Row>
  <Row
    figure={108.95}
    title="Est. total"
    total={true}
  />
  <ExpandableItem
    closePrefix="Hide"
    openPrefix="See"
    title="item details"
  >
    <ItemDetails
      itemName="OFM Essentials Racecar-Style Leather Gaming Chair, Green"
      pictureURL="https://i5.walmartimages.com/asr/79916c35-3293-48ff-acff-be325b2e4c84_2.ef5b1bd1451de6b63a7c20d2485d56c6.jpeg?odnHeight=100&odnWidth=100&odnBg=FFFFFF"
      prevPrice={102.96}
      price={99.11}
      quantity={1}
    />
  </ExpandableItem>
  <ExpandableItem
    closePrefix="Hide"
    openPrefix="Apply"
    title="promo code"
  >
    <DiscountForm
      url="dotted://purchaseData"
    />
  </ExpandableItem>
</div>
`;
